### PR TITLE
Add CEA weekly test config

### PR DESCRIPTION
**Pull request by pzehner**
This PR aims to extend testing by running Kokkos tests on a weekly basis on the Ruche local cluster, by CEA means. The configuration is minimal for now and consists in a debug and release builds for the Cuda backend, on A100 GPUs.

The workflow is scheduled to run every Saturday at 2 am UTC, or manually.

The self-hosted runner on Ruche checks if the job can be executed. Scheduled runs can only take place on weekend. Manual runs have to be triggered by a white-listed user (I prefer to discuss about the members of the list on nucleus). No other job types (on push, etc.) are allowed.

Registration of the self-hosted runner was done with @crtrott.

The same configuration was tested on a dummy project, so it should run on the first try here...

### CODE REVIEW

File: .github/workflows/weekly-cea.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            flags: -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_AMPERE80=ON
+            modules: gcc/11.2.0/gcc-4.8.5 cuda/12.2.1/gcc-11.2.0 cmake/3.28.3/gcc-11.2.0
+            queue: gpua100
+            runner: [self-hosted, cuda]
+
+    runs-on: ${{ matrix.backend.runner }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```
Comment (romintomasetti) [inline .github/workflows/weekly-cea.yml:26]:
```suggestion
      - uses: actions/checkout@v4
```
With this you already get meaningful step summary:
image

File: .github/workflows/weekly-cea.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            -DCMAKE_CXX_STANDARD=17 \
+            -DCMAKE_CXX_FLAGS=-Werror \
+            -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
+            -DKokkos_ENABLE_TESTS=ON \
+            -DKokkos_ENABLE_EXAMPLES=ON \
+            ${{ matrix.backend.flags }}
+
+      - name: Build
+        run: |
+          module load ${{ matrix.backend.modules }}
```
Comment (romintomasetti) [inline .github/workflows/weekly-cea.yml:42]: Are you sure you need to load the modules at each job step ? I think you run them all in the same process, so I guess you could load them once in a dedicated step before the `Configure` step. Module loading failures would be easily identified then.

File: .github/workflows/weekly-cea.yml (hunk 1 above)
Comment (pzehner) [reply .github/workflows/weekly-cea.yml:26]: Right.

File: .github/workflows/weekly-cea.yml (hunk 2 above)
Comment (pzehner) [reply .github/workflows/weekly-cea.yml:42]: Job steps are executed on the same runner, but [on different processes](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idsteps), so I cannot load modules for once. It's still possible to set environment variables from one step to the other, but [the process is tedious and has to be done for each of them](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable).

File: .github/workflows/weekly-cea.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+name: Weekly CEA CI
```
Comment (dalg24) [inline .github/workflows/weekly-cea.yml:1]:
```suggestion
name: Weekly CEA builds
```
Technically this is not Continuous Integration :)

**Review by dalg24 — APPROVED** [review]

File: .github/workflows/weekly-cea.yml (hunk 3 above)
Comment (pzehner) [reply .github/workflows/weekly-cea.yml:1]: Right!

**Review by masterleinad — APPROVED** [review]
Let's give it a shot!

**Pull request merged**